src/app/loadout/ModPicker.tsx

```diff
@@ -261,3 +228,62 @@ function ModPicker({ plugSets, lockedMods, initialQuery, minHeight, onAccept, on
 }
 
 export default connect<StoreProps>(mapStateToProps)(ModPicker);
+
+/**
+ * Determine whether an armor mod can still be selected, given that the `selected` mods have already been selected.
+ * This doesn't take into account the actual armor that's in the loadout and what slots it has.
+ */
+function isModSelectable(
+  mod: PluggableInventoryItemDefinition,
+  selected: PluggableInventoryItemDefinition[]
+) {
+  const { plugCategoryHash, energyCost } = mod.plug;
+  const isSlotSpecificCategory = slotSpecificPlugCategoryHashes.includes(plugCategoryHash);
+
+  // Already selected mods that are in the same category as "mod"
+  let associatedLockedMods: PluggableInventoryItemDefinition[] = [];
+
+  if (isSlotSpecificCategory || plugCategoryHash === armor2PlugCategoryHashesByName.general) {
+    // General and slot-specific mods just match to the same category hash
+    associatedLockedMods = selected.filter((mod) => mod.plug.plugCategoryHash === plugCategoryHash);
+  } else if (raidModPlugCategoryHashes.includes(plugCategoryHash)) {
+    // Raid mods match to any other raid mod, since a single armor piece can only have one raid mod slot
+    associatedLockedMods = selected.filter((mod) =>
+      raidModPlugCategoryHashes.includes(mod.plug.plugCategoryHash)
+    );
+  } else {
+    // This is some unknown/unmapped mod slot, match all other unknown mod slots
+    associatedLockedMods = selected.filter(
+      (mod) => !knownModPlugCategoryHashes.includes(mod.plug.plugCategoryHash)
+    );
+  }
+
+  // Slot-specific mods (e.g. chest mods) can slot 2 per piece, so make sure the sum of energy doesn't
+  // exceed the maximum and that energy all aligns. This doesn't check other mods that could be on the
+  // item because we haven't assigned those to specific pieces.
+  // TODO: This also doesn't check whether we add 5 1-cost chest mods?
```

Comment:
Did we intend to include a check for maxSelectable here, or is that handled elsewhere? If the latter, why does this check `associatedLockedMods.length < MAX_SLOT_INDEPENDENT_MODS`?

Reply:
Handled elsewhere now, that `maxSelectable` prop handles it. See[this](https://github.com/DestinyItemManager/DIM/blob/master/src/app/loadout/plug-drawer/PlugSection.tsx#L94).